--- source/game.cpp	2018-05-30 09:03:24.322621006 +0200
+++ build/game.cpp	2018-05-30 09:14:21.977606904 +0200
@@ -2942,11 +2942,11 @@
 		//user_width = 1184;
 		//user_height = 720;
 
-		if( user_width >= 4*default_width_c ) {
+		/*if( user_width >= 4*default_width_c ) {
 			scale_width = 4.0f;
 			LOG("scale width 4x\n");
 		}
-		else if( user_width >= 3*default_width_c ) {
+		else*/ if( user_width >= 3*default_width_c ) {
 			scale_width = 3.0f;
 			LOG("scale width 3x\n");
 		}
@@ -2963,11 +2963,11 @@
 			return false;
 		}
 
-		if( user_height >= 4*default_height_c ) {
+		/*if( user_height >= 4*default_height_c ) {
 			scale_height = 4.0f;
 			LOG("scale height 4x\n");
 		}
-		else if( user_height >= 3*default_height_c ) {
+		else*/ if( user_height >= 3*default_height_c ) {
 			scale_height = 3.0f;
 			LOG("scale height 3x\n");
 		}
@@ -3022,11 +3022,11 @@
 		screen = new Gigalomania::Screen();
 
 #if SDL_MAJOR_VERSION == 1
-		if( screen->open(4*default_width_c, 4*default_height_c, fullscreen) ) {
+		/*if( screen->open(4*default_width_c, 4*default_height_c, fullscreen) ) {
 			scale_width = scale_height = 4.0f;
 			LOG("scale 4x\n");
 		}
-		else if( screen->open(3*default_width_c, 3*default_height_c, fullscreen) ) {
+		else*/ if( screen->open(3*default_width_c, 3*default_height_c, fullscreen) ) {
 			scale_width = scale_height = 3.0f;
 			LOG("scale 3x\n");
 		}
@@ -5250,8 +5250,8 @@
 #ifdef _DEBUG
 	fullscreen = false;
 #endif
-	//debugwindow = true;
-	//fullscreen = false;
+	debugwindow = true;
+	fullscreen = false;
 
 #if !defined(__ANDROID__)
     // n.b., crashes when run on Galaxy Nexus (even though fine in the emulator)
