diff -ru source/regress/Makefile.am source-new/regress/Makefile.am
--- source/regress/Makefile.am	2017-08-14 03:18:27.000000000 -0700
+++ source-new/regress/Makefile.am	2017-11-16 22:56:38.912996868 -0800
@@ -9,23 +9,6 @@
 	fseek \
 	tryopen
 
-if !WIN32_HOST
-pkglib_LTLIBRARIES=	malloc.la nonrandomopen.la
-
-malloc_la_SOURCES=	malloc.c
-malloc_la_LDFLAGS=	-module -avoid-version
-malloc_la_LIBADD=	${DLSYM_LIB}
-
-nonrandomopen_la_SOURCES=	nonrandomopen.c
-nonrandomopen_la_LDFLAGS=	-module -avoid-version
-nonrandomopen_la_LIBADD=	${DLSYM_LIB}
-
-# we have to do it this way, because if we choose noinst_LTLIBRARIES,
-# only the .a files are built
-install-pkglibLTLIBRARIES:
-	@echo not installing shared objects needed only for testing
-endif
-
 EXTRA_DIST= \
 	CMakeLists.txt \
 	NiHTest.pm \
